libs/backend/src/routes/user/[username]/activity/index.ts

```diff
@@ -0,0 +1,40 @@
+import Puzzle from "@/models/puzzle/puzzle.js";
+import Submission from "@/models/submission/submission.js";
+import User from "@/models/user/user.js";
+import { FastifyInstance } from "fastify";
+import { userEntitySchema } from "types";
+
+export default async function userByUsernameActivityRoutes(fastify: FastifyInstance) {
+	fastify.get("/", async (request, reply) => {
+		const parseResult = userEntitySchema.pick({ username: true }).safeParse(request.params);
+
+		if (!parseResult.success) {
+			return reply.status(400).send({ message: "Invalid request data" });
```

Comment:
message seems fairly generic, could possibly make a `const` somewhere to only declare it once :+1:

Reply:
started on this, the under taking is too big for one evening, would rather get through the other issues as well, since this is something generic and probably needs to be taken on in a more systematic manner in the future